src/lib/pickles/pickles.ml

```diff
@@ -2998,244 +2998,322 @@ module Make_str (_ : Wire_types.Concrete) = struct
-        ; main =
-            (fun { public_input = () } ->
-              let dummy_proof =
-                As_prover.Ref.create (fun () ->
-                    Proof0.dummy Nat.N2.n Nat.N2.n Nat.N2.n ~domain_log2:15 )
-              in
-              { previous_proof_statements =
-                  [ { public_input = ()
-                    ; proof = dummy_proof
-                    ; proof_must_verify = Boolean.false_
-                    }
-                  ; { public_input = ()
-                    ; proof = dummy_proof
-                    ; proof_must_verify = Boolean.false_
-                    }
-                  ]
-              ; public_output = ()
-              ; auxiliary_output = ()
-              } )
-        ; uses_lookup = false
-        }
+        module A = Statement
+        module A_value = Statement
+
+        let typ = Typ.unit
+
+        module Branches = Nat.N1
+        module Max_proofs_verified = Nat.N2
+
+        let constraint_constants : Snark_keys_header.Constraint_constants.t =
+          { sub_windows_per_window = 0
+          ; ledger_depth = 0
+          ; work_delay = 0
+          ; block_window_duration_ms = 0
+          ; transaction_capacity = Log_2 0
+          ; pending_coinbase_depth = 0
+          ; coinbase_amount = Unsigned.UInt64.of_int 0
+          ; supercharged_coinbase_factor = 0
+          ; account_creation_fee = Unsigned.UInt64.of_int 0
+          ; fork = None
+          }
 
-      let override_wrap_main =
-        { wrap_main =
-            (fun _ _ _ _ _ _ _ ->
-              let requests =
-                (* The requests that the logic in [Wrap.wrap] use to pass
-                   values into and out of the wrap proof circuit.
-                   Since these are unnecessary for the dummy circuit below, we
-                   generate them without using them.
-                *)
-                Requests.Wrap.create ()
-              in
-              (* A replacement for the 'wrap' circuit, which makes no
-                 assertions about the statement that it receives as its first
-                 argument.
-              *)
-              let wrap_main _ =
-                let module SC' = SC in
-                let open Impls.Wrap in
-                let open Wrap_main_inputs in
-                let open Wrap_main in
-                (* Create some variables to be used in constraints below. *)
-                let x =
-                  exists Field.typ ~compute:(fun () -> Field.Constant.of_int 3)
-                in
-                let y =
-                  exists Field.typ ~compute:(fun () -> Field.Constant.of_int 0)
+        let rule self : _ Inductive_rule.t =
+          { identifier = "main"
+          ; prevs = [ self; self ]
+          ; main =
+              (fun { public_input = () } ->
+                let dummy_proof =
+                  As_prover.Ref.create (fun () ->
+                      Proof0.dummy Nat.N2.n Nat.N2.n Nat.N2.n ~domain_log2:15 )
```

Comment:
I had to modify this one as well (domain_log2) to trigger the bug

Reply:
<img width="538" alt="Screenshot 2023-02-08 at 4 43 33 PM" src="https://user-images.githubusercontent.com/1316043/217685195-e4edb96e-b216-4659-8ef5-1cc35e96aa4f.png">

Reply:
so my functor as expecting this additional argument: 

<img width="515" alt="Screenshot 2023-02-08 at 4 44 00 PM" src="https://user-images.githubusercontent.com/1316043/217685236-8d2337ba-c9d5-4282-abf5-921c3373dc07.png">